### PR TITLE
Launch core integration part 2

### PR DESCRIPTION
**Pull request by ktangsali**
# Modulus Pull Request

## Description
This brings the weather, CFD and Healthcare examples from Launch repo. 

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/modulus/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [x] The [CHANGELOG.md](https://github.com/NVIDIA/modulus/blob/main/CHANGELOG.md) is up to date with these changes.
- [x] An [issue](https://github.com/NVIDIA/modulus/issues) is linked to this pull request.

### CODE REVIEW

File: examples/.markdownlint.yaml [hunk 1]
```diff
@@ -0,0 +1,274 @@
+# Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```
Comment (akshaysubr) [inline examples/.markdownlint.yaml:1]: How is this going to be used in CI? Will markdownlint use the first yaml it finds going up the directory tree from a file?

**Review by akshaysubr — APPROVED** [review]

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -29,7 +29,7 @@ repos:
           "--ignore-magic", "--fail-under=99", "--exclude=['setup.py', 'test', 'build', 'docs']",  
           "--ignore-regex=['forward', 'backward', 'reset_parameters', 'extra_repr', 'MetaData', 'apply_activation','exec_activation']", 
           "--color", "--"]
-      exclude: ^modulus/internal/
+      exclude: ^modulus/internal/|^examples/
```
[the remark this answers is not in the document]
Comment (ktangsali) [reply .pre-commit-config.yaml]: Great point. I was on the fence about this, but I guess it makes sense to impose the requirement. Just removed the change to interrogate section and added the missing docs

File: examples/.markdownlint.yaml (hunk 1 above)
Comment (ktangsali) [reply examples/.markdownlint.yaml:1]: Actually, the CI will just look at the top level yaml. To run the yaml from examples, one needs to explicitly run markdownlint in examples folder. I'd say, to limit the scope, let's consider introducing that to CI in a separate PR.

File: examples/.markdownlint.yaml [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+#
```
Comment (NickGeneva) [inline examples/.markdownlint.yaml:2]: Am I missing something or is the general index readme for the examples missing? Is that intentional?

File: examples/.markdownlint.yaml [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+# Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (NickGeneva) [inline examples/.markdownlint.yaml:3]: Remember to rebase this, do not squash

**Review by NickGeneva — APPROVED** [review]

File: examples/.markdownlint.yaml (hunk 3 above)
Comment (ktangsali) [reply examples/.markdownlint.yaml:2]: That is intentional. With the discussion around a master README, I realized it's better to bring the README in a separate PR.

File: examples/.markdownlint.yaml (hunk 4 above)
Comment (ktangsali) [reply examples/.markdownlint.yaml:3]: Sure thing!

**Pull request merged**